### PR TITLE
Update experiment description for the SP paper

### PR DESCRIPTION
**Pull request by ywcui1990**
@lscheinkman Please refer to experimentDescription.md to reproduce results for the SP paper.

### CODE REVIEW

File: projects/sp_paper/experimentDescriptions.md [hunk 1]
```diff
@@ -87,8 +101,8 @@ Train faulty_SP on random bar set dataset
 	`python python_sp_tm_model.py --trainSP 1 --boostStrength 0`
 * Run with learning SP, and boosting
  
-`python python_sp_tm_model.py --trainSP 1 --boostStrength 20`
-* plot results
-
-# MNIST experiment
-`python train_sp.py -t 1 -d mnist --spatialImp py -e 2 -b 1`
\ No newline at end of file
+	`python python_sp_tm_model.py --trainSP 1 --boostStrength 20`
```
Comment (lscheinkman) [inline projects/sp_paper/experimentDescriptions.md]: did you mean `run_sp_tm_model.py`

File: projects/sp_paper/experimentDescriptions.md [hunk 2]
```diff
@@ -33,42 +33,56 @@
...
 
 * Random bar sets (more than two random bars per input)
 
 `python train_sp.py -t 1 -d randomBarSets -b 1 --name random_bars_with_topology \
 --spatialImp monitored_sp --changeDataSetContinuously 1 --boosting 1`
 
+# MNIST experiment
+Fig. 4C
+
+`python train_sp.py -t 1 -d mnist --spatialImp py -e 2 -b 1`
```
Comment (lscheinkman) [inline projects/sp_paper/experimentDescriptions.md]: This script fails with:
```
OSError: [Errno 2] No such file or directory: './htmresearch/projects/sp_paper/data/mnist/training'
```

File: projects/sp_paper/experimentDescriptions.md [hunk 3]
```diff
@@ -87,8 +101,8 @@ Train faulty_SP on random bar set dataset
...
-`python python_sp_tm_model.py --trainSP 1 --boostStrength 20`
-* plot results
-
-# MNIST experiment
-`python train_sp.py -t 1 -d mnist --spatialImp py -e 2 -b 1`
\ No newline at end of file
+	`python python_sp_tm_model.py --trainSP 1 --boostStrength 20`
+	
+* plot results (Fig. 6)
+	`python plot_nyc_taxi_experiment.py`
```
Comment (lscheinkman) [inline projects/sp_paper/experimentDescriptions.md]: did you mean `plot_nyc_taxi_performance.py`

**Comment by lscheinkman** [comment]
run_sp_tm_model.py script is failing with:

```
Traceback (most recent call last):
  File "run_sp_tm_model.py", line 33, in <module>
    from nupic.frameworks.opf.predictionmetricsmanager import MetricsManager
ImportError: No module named predictionmetricsmanager
```

**Comment by ywcui1990** [comment]
@lscheinkman I fixed the experiment for Fig. 6 (run_sp_tm_model) and added instruction to download the mnist. Please review again.

**Comment by lscheinkman** [comment]
@ywcui1990 The **projects/sp_paper/data/nyc_taxi.csv** file was deleted by this PR however it is used as input for `plot_nyc_taxi_performance.py`

**Comment by ywcui1990** [comment]
@lscheinkman  I fixed Fig. 6C. Please review.

**Comment by ywcui1990** [comment]
@lscheinkman I also fixed the MNIST experiment (Fig. 4C) You should be able to generate the result with `python train_sp.py -t 1 -d mnist --spatialImp py -e 2 -b 1 --showExampleRFs 1`

**Pull request merged**